Mathlib/LinearAlgebra/FiniteDimensional.lean

```diff
@@ -54,22 +54,29 @@ theorem finrank_add_le_finrank_add_finrank (s t : Submodule K V) [FiniteDimensio
   rw [← finrank_sup_add_finrank_inf_eq]
   exact self_le_add_right _ _
 
+theorem finrank_add_finrank_le_of_disjoint [FiniteDimensional K V]
+    {s t : Submodule K V} (hdisjoint : Disjoint s t) :
+    finrank K s + finrank K t ≤ finrank K V := by
+  rw [← Submodule.finrank_sup_add_finrank_inf_eq s t, hdisjoint.eq_bot, finrank_bot, add_zero]
+  exact Submodule.finrank_le _
+
 theorem eq_top_of_disjoint [FiniteDimensional K V] (s t : Submodule K V)
-    (hdim : finrank K s + finrank K t = finrank K V) (hdisjoint : Disjoint s t) : s ⊔ t = ⊤ := by
+    (hdim : finrank K V ≤ finrank K s + finrank K t) (hdisjoint : Disjoint s t) : s ⊔ t = ⊤ := by
   have h_finrank_inf : finrank K ↑(s ⊓ t) = 0 := by
     rw [disjoint_iff_inf_le, le_bot_iff] at hdisjoint
     rw [hdisjoint, finrank_bot]
   apply eq_top_of_finrank_eq
-  rw [← hdim]
+  replace hdim : finrank K V = finrank K s + finrank K t :=
+    le_antisymm hdim (finrank_add_finrank_le_of_disjoint hdisjoint)
+  rw [hdim]
   convert s.finrank_sup_add_finrank_inf_eq t
   rw [h_finrank_inf]
   rfl
 
-theorem finrank_add_finrank_le_of_disjoint [FiniteDimensional K V]
-    {s t : Submodule K V} (hdisjoint : Disjoint s t) :
-    finrank K s + finrank K t ≤ finrank K V := by
-  rw [← Submodule.finrank_sup_add_finrank_inf_eq s t, hdisjoint.eq_bot, finrank_bot, add_zero]
-  exact Submodule.finrank_le _
+theorem disjoint_iff_isCompl [FiniteDimensional K V] (s t : Submodule K V)
+    (hdim : finrank K V ≤ finrank K s + finrank K t) :
+    IsCompl s t ↔ Disjoint s t :=
```

Comment:
Please swap either the name components or the two sides of the iff.

Reply:
Eek, thanks: renamed.